material & home page update (#123)
codelingo/docs: docs/index.md

# Overview

> Codelingo is a platform for software development teams to store, discuss and
> code their tacit knowledge into project specific rules.

Reviewer: ```suggestion
CodeLingo is a platform for software development teams to store, discuss, and
codify their tacit knowledge into project specific rules.
```

Reviewer: Thoughts on 'rules' vs 'Rules'? I feel like capitalising our own nomenclature helps to highlight their significance in sentences. Also, up to you on the Oxford comma ;)

Reviewer: Okay, after reading more, it seems you _are_ capitalising. There's just a few that you haven't. Can you please make those all consistent?

> Our platform allows development teams to capture a discussion in a Pull Request,
> create a Rule around that discussion then use our DSL, CLQL to generate a
> project specific rule that is used to automatically review incoming code to your codebase.

Reviewer: ```suggestion
Our platform allows development teams to capture a discussion in a Pull Request,
create a Rule around that discussion then use our DSL (CLQL: CodeLingo Query Language), to generate a
project specific rule that is used to automatically review incoming code to your codebase.
```

## Dashboard

> The CodeLingo Dashboard
> [(codelingo.io/dashboard)](https://www.codelingo.io/dashboard) lets you manage
> your rules for each of your repositories. It is your teams hub for discussion and creating project specific rules.

Reviewer: ```suggestion
The [CodeLingo Dashboard](https://dash.codelingo.io) lets you manage your rules for each of your repositories. It is your teams hub for discussing, creating, editing, and testing project specific rules.
```

Reviewer 2: s/your teams hub/your team's hub/

## Writing Rule Queries

> Using CodeLingo Query Language (CLQL) you can write queries for your rules that
> will capture anti patterns in your codebase.

Reviewer: ```suggestion
Using CodeLingo Query Language (CLQL) you can write queries for your rules that will capture anti-patterns in your codebase.
```

## Rules

A Rule is an encoded project specific best practice used to guide development. A Rule can be used for: coding styles, performance tuning, security audits, debugging, avoiding gotchas, reducing complexity and churn, and more.

> Rules can be created from the dashboard, you can either create a blank Rule or
> one from a Rule Template.

Reviewer: ```suggestion
Rules can be created from the dashboard. You can either create a blank Rule or one from a Rule Template.
```
